x-pack/plugins/ml/public/application/data_frame_analytics/common/analytics.ts

```diff
@@ -414,19 +422,37 @@ export const useRefreshAnalyticsList = (
 
 const DEFAULT_SIG_FIGS = 3;
 
-export function getValuesFromResponse(response: RegressionEvaluateResponse) {
-  let meanSquaredError = response?.regression?.mse?.value;
+interface RegressionEvaluteExtractedResponse {
+  mse: number | string;
+  msle: number | string;
+  huber: number | string;
+  r_squared: number | string;
+}
 
-  if (meanSquaredError) {
-    meanSquaredError = Number(meanSquaredError.toPrecision(DEFAULT_SIG_FIGS));
-  }
+export const EMPTY_STAT = '--';
 
-  let rSquared = response?.regression?.r_squared?.value;
-  if (rSquared) {
-    rSquared = Number(rSquared.toPrecision(DEFAULT_SIG_FIGS));
+export function getValuesFromResponse(response: RegressionEvaluateResponse) {
+  const results: RegressionEvaluteExtractedResponse = {
+    mse: EMPTY_STAT,
+    msle: EMPTY_STAT,
+    huber: EMPTY_STAT,
+    r_squared: EMPTY_STAT,
+  };
+
+  if (response?.regression) {
+    for (const statType in response.regression) {
```

Comment:
rather than using a for..of, this loop could be written as:
```
Object.entries(response.regression).forEach(([statType, { value }]) => {
  results[statType as keyof RegressionEvaluteExtractedResponse] = value.toPrecision(
    DEFAULT_SIG_FIGS
  );
});
```
but i don't know whether it's more readable :)

Reply:
I prefer the more readable for loop option since in this case there are only 4 properties we're checking for.